modules/host/contractcreation.go

```
@@ -54,68 +66,60 @@ func (h *Host) considerContract(t consensus.Transaction) (updatedTransaction con
 		err = errors.New("burn payout needs to go to the empty address")
 		return
 	}
-	// Verify that output for failed proofs matches burn.
-	maxBurn := h.announcement.Burn * consensus.Currency(fileSize) * consensus.Currency(t.FileContracts[0].ChallengeWindow)
-	if t.FileContracts[0].MissedProofPayout > maxBurn {
-		err = errors.New("burn payout is too high for a missed proof.")
-		return
-	}
+
 	// Verify that the contract fund covers the payout and burn for the whole
 	// duration.
-	requiredFund := (h.announcement.Price*consensus.Currency(fullDuration) + h.announcement.Burn*consensus.Currency(contractDuration)) * consensus.Currency(fileSize)
-	if t.FileContracts[0].ContractFund < requiredFund {
-		err = errors.New("ContractFund does not cover the entire duration of the contract.")
+	requiredFund := (h.announcement.Price + h.announcement.Burn) * consensus.Currency(duration) * consensus.Currency(fileSize)
+	if contract.Payout != requiredFund {
+		err = errors.New("ContractFund does not match the terms of service.")
 		return
 	}
 
 	// Add enough funds to the transaction to cover the penalty half of the
 	// agreement.
-	penalty := h.announcement.Burn * consensus.Currency(fileSize) * consensus.Currency(contractDuration)
+	penalty := h.announcement.Burn * consensus.Currency(fileSize) * consensus.Currency(duration)
 	id, err := h.wallet.RegisterTransaction(t)
 	if err != nil {
+		err = HostCapacityErr // hide the fact that the host is having wallet issues.
 		return
 	}
 	err = h.wallet.FundTransaction(id, penalty)
 	if err != nil {
-		// TODO: This leaks that the host is out of money.
+		err = HostCapacityErr // hide the fact that the host is having wallet issues.
 		return
 	}
 	updatedTransaction, err = h.wallet.SignTransaction(id, true)
+	if err != nil {
+		err = HostCapacityErr // hide the fact that the host is having wallet issues.
+		return
+	}
+
+	// Update the amount of space the host has for sale.
+	h.spaceRemaining -= int64(fileSize)
```

yeah, I would just return an error. Maybe add a "-f" option if you want to set it anyway.